### PR TITLE
extra unit test on keyword behaviour

### PR DESCRIPTION
**Pull request by dorchard**
Adds a unit test on the behaviour of `launch.m` with partial data tiles missing some of the keywords. This uses the `stdout` of the MATLAB engine.

### CODE REVIEW

File: tests/unit_tests/test_ustar_cp/test_launch.py [hunk 1]
```diff
@@ -129,7 +129,59 @@ def test_launch_empty_folder(setup_test_environment, matlab_engine):
 
     # Run the MATLAB function
     exitcode = eng.launch(input_folder, output_folder)
-    
+
     # Check that the exitcode indicates an error
     assert exitcode == 0, "Expected zero exitcode for empty input folder."
 
+def test_missing_keywords(setup_test_environment, matlab_engine, extract_section_between_keywords):
```
Comment (j-emberton) [inline tests/unit_tests/test_ustar_cp/test_launch.py]:
```suggestion
def test_missing_keywords(setup_test_environment, matlab_engine):
```

**Comment by j-emberton** [comment]
Did you base this branch off 'main' rather than test_launch.m?

It's slightly out of sync with my test_launch.m branch which you just reviewed. E.g. setup_folders fixture is now called setup_test_environment.

Otherwise it looks ok

**Comment by dorchard** [comment]
Hm. This is based off `[refactor_and_test_launch](https://github.com/Cambridge-ICCS/ONEFlux/tree/refactor_and_test_launch)`. I'm using `setup_test_environment`, which I think is the most up-to-date name right?

File: tests/unit_tests/test_ustar_cp/test_launch.py [hunk 2]
```diff
@@ -129,7 +129,59 @@ def test_launch_empty_folder(setup_test_environment, matlab_engine):
 
     # Run the MATLAB function
     exitcode = eng.launch(input_folder, output_folder)
-    
+
     # Check that the exitcode indicates an error
     assert exitcode == 0, "Expected zero exitcode for empty input folder."
 
+def test_missing_keywords(setup_test_environment, matlab_engine):
```
Comment (j-emberton) [inline tests/unit_tests/test_ustar_cp/test_launch.py:136]:
```suggestion
def test_missing_keywords(setup_folders matlab_engine):
```

**Comment by j-emberton** [comment]
> Hm. This is based off `[refactor_and_test_launch](https://github.com/Cambridge-ICCS/ONEFlux/tree/refactor_and_test_launch)`. I'm using `setup_test_environment`, which I think is the most up-to-date name right?

Sorry I got myself completely confused there, mixing up branch and PR names and all sorts. I think you're all good.

**Pull request merged**